sd/consul/instancer.go

```diff
@@ -70,6 +73,8 @@ func (s *Instancer) loop(lastIndex uint64) {
 			s.cache.Update(sd.Event{Err: err})
 		default:
 			s.cache.Update(sd.Event{Instances: instances})
+			time.Sleep(d)
+			d = conn.Exponential(d)
```

Comment:
Sorry for not catching this earlier. We want to implement an upper limit e.g. 30s, and have a successful reconnect reset the `d` value to its default state.